src/Legacy/DynamoRevitVersionSelector/VersionSelectorApp.cs

```diff
@@ -109,104 +95,125 @@ public Result OnStartup(UIControlledApplication application)
-                data.WriteToRegistry();
-
-                splitButton.Enabled = false; //Disable the split button, no more needed.
-                splitButton.Visible = false; //Hide it from the UI
-                //For older than 0.8.2, hide the Dynamo Versions ribbon panel
-                //Otherwise rename it to "Visual Programming", so that DynamoRevit
-                //will add it's button to this panel.
-                if (p.VersionInfo < new Version(0, 8, 2))
-                {
-                    ribbonPanel.Visible = false; //Hide the panel
-                }
-                else
-                {
-                    ribbonPanel.Name = Resources.VisualProgramming;//Same as used in App Description for DynamoRevitApp
-                    ribbonPanel.Title = Resources.VisualProgramming;
-                }
-
-                //Initialize application
-                var ass = Assembly.LoadFrom(path);
-                var revitApp = ass.CreateInstance("Dynamo.Applications.DynamoRevitApp");
-                revitApp.GetType()
-                    .GetMethod("OnStartup")
-                    .Invoke(revitApp, new object[] { uiApplication });
-
-                //Invoke command
-                string message = string.Empty;
-                var revitCmd = ass.CreateInstance("Dynamo.Applications.DynamoRevit");
-                revitCmd.GetType()
-                    .GetMethod("Execute")
-                    .Invoke(revitCmd, new object[] { commandData, message, null });
-
-                uiApplication = null; //release application, no more needed.
-                return true;
-            }
-            catch (Exception)
-            {
-                splitButton.Enabled = true; //Ensure that the split button is enabled.
-                splitButton.Visible = true; //Also the split button is visible
-                ribbonPanel.Visible = true; //As well as the panel is visible
-                throw;
-            }
+            var product = PromptVersionSelectorDialog(Products);
+            if(product.HasValue)
+                LaunchDynamoCommand(product.Value, e);
         }
 
-        private SplitButton AddSplitButtonGroup(RibbonPanel panel)
+        void beforeExecuted(object sender, BeforeExecutedEventArgs e)
         {
-            var versionData = new SplitButtonData("versions", Resources.DynamoVersions);
-            var button = panel.AddItem(versionData) as SplitButton;
-
-            Bitmap dynamoIcon = Resources.dynamo_32x32;
+            e.UsingCommandData = true;
+        }
 
-            BitmapSource bitmapSource =
-                Imaging.CreateBitmapSourceFromHBitmap(
-                    dynamoIcon.GetHbitmap(),
-                    IntPtr.Zero,
-                    Int32Rect.Empty,
-                    BitmapSizeOptions.FromEmptyOptions());
+        void canExecute(object sender, CanExecuteEventArgs e)
+        {
+            e.CanExecute = e.ActiveDocument != null;
```

Comment:
So I agree that the logic here may not be the same with SetRunEnabledBasedOnContext depending on the workflow.